src/server/zset_family.cc

```diff
@@ -1058,6 +1058,38 @@ struct SetOpArgs {
   bool with_scores = false;
 };
 
+void HandleOpStatus(ConnectionContext* cntx, OpStatus op_status) {
+  switch (op_status) {
+    case OpStatus::INVALID_FLOAT:
+      return (*cntx)->SendError("weight value is not a float", kSyntaxErrType);
+    default:
+      return (*cntx)->SendError(op_status);
+  }
+}
+
+OpResult<ScoredMap> IntersectResults(const vector<OpResult<ScoredMap>>& results, AggType agg_type) {
+  OpResult<ScoredMap> result;
+  for (auto op_res : results) {
```

Comment:
Please use `const auto&` in order to avoid a copy of the value (map of strings in this case)

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/00842a80fcc12b92febd0c2330b4a08882c56722#diff-294d1289b1741ae808c06f38d260077a53dcba7762c845f9677a068568821fc9R1070-R1072